### PR TITLE
Add average time to initial response metric

### PR DESCRIPTION
**Pull request by jpaidoussi**
Record metric of minutes elapsed until initial response to a case

### CODE REVIEW

**Comment by jpaidoussi** [comment]
@rudigiesler @KaitCrawford @smn Ready for review

File: casepro/cases/tests.py [hunk 1]
```diff
@@ -1129,14 +1129,15 @@ def test_list(self):
         ]})
 
         response = self.url_get('unicef', url + '?with_activity=1', HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.maxDiff = None
```
Comment (rudigiesler) [inline casepro/cases/tests.py]: Should probably remove this line?

File: casepro/cases/views.py [hunk 2]
```diff
@@ -390,6 +391,14 @@ def post(self, request, *args, **kwargs):
     class List(OrgPermsMixin, SmartListView):
         paginate_by = None
 
+        @staticmethod
+        def humanise_minutes(minutes):
+            if minutes < 60:
+                return "%.0fm" % ceil(minutes)
+            else:
+                hrs, mins = divmod(minutes, 60)
+                return "%.fh %.0fm" % (hrs, ceil(mins))
```
Comment (rudigiesler) [inline casepro/cases/views.py]: Maybe we should also allow for days? I'm not sure what the expected average response times are supposed to be, but converting 112h -> 4d 16h is not really something that's easy to do in your head.

Comment (jpaidoussi) [inline casepro/cases/views.py]: Good idea. I've updated the function to include days

**Comment by jpaidoussi** [comment]
@rudigiesler Ready for review

**Comment by jpaidoussi** [comment]
@rowanseymour Ready for you to look over

**Comment by rudigiesler** [comment]
:+1:

File: casepro/statistics/migrations/0008_dailyminutetotalcount.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    operations = [
+        migrations.CreateModel(
+            name='DailyMinuteTotalCount',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('item_type', models.CharField(help_text='The thing being counted', max_length=1)),
+                ('scope', models.CharField(help_text='The scope in which it is being counted', max_length=32)),
+                ('count', models.IntegerField()),
+                ('minutes', models.IntegerField()),
```
Comment (rowanseymour) [inline casepro/statistics/migrations/0008_dailyminutetotalcount.py]: Bit late now but could have used seconds. Never any harm in erring on the side of accuracy

Comment (rudigiesler) [inline casepro/statistics/migrations/0008_dailyminutetotalcount.py]: We could maybe switch it to seconds? I just would have thought that responding/closing cases would be more on the scale of minutes than seconds. We also need to be careful about maxing out the `IntegerField`, but I don't think a 60x factor will make a big difference with that (we definitely don't want milliseconds though, that would be a 60000x factor)

Comment (rowanseymour) [inline casepro/statistics/migrations/0008_dailyminutetotalcount.py]: I agree that cases are unlikely to be opened and closed within a time frame that requires more accuracy than minutes - but if we want to re-use this class for timing other things we might need more accuracy. If you were timing bulk replies for example, you wouldn't want 59 seconds becoming 0 minutes etc.

As you say I don't think 60x factor changes much here - if we think we're anywhere near overflowing we should switch to something like `BigIntegerField`. Anyways, your call.

Comment (rudigiesler) [inline casepro/statistics/migrations/0008_dailyminutetotalcount.py]: I think due to time restrains we'll keep it as is, and take the technical debt of having to add/change it to seconds if we need to measure something more accurately in the future.

**Pull request merged**